openmsupply/application-manager-web-app
# #577B Actions triggered on Re-assignment

## Linked PRs
- Back-end https://github.com/openmsupply/application-manager-server/pull/586
- Templates to use `dev.current` on branch `reassignment-changes` in application-manager-templates

### Changes
- When selecting option "Re-assign" in UI a second dropdown shows for selecting the new reviewer
- After selecting the next reviewer the reassignment is completed by Front-end
- Feature to change re-assignment already implemented partially in #982. Just added the ability to disable the previous review.status linked to the un-assigned reveiwer to have review.status = DISCONTINUED (action that runs ON_REVIEW_UNASSIGN)
- Also allows to revert that by setting an existing review associated with the new ReviewAssignment assigned to have review.status = DRAFT.

### Testing
- Using `dev` snapshot, login as `testAssigner1` and check review home page of application `S-MAS-0004`
  - you can see it's assigned to `assessor1` which is disabled since already submitted
  - and `assessor2` can be re-assigned
-  Log in as `assessor2` and start a review, just add one review to one of the responses
- Log in again as `testAssigner1` and select to reassign section 2 from `assessor2` to `Reviewer2`
  - This would have blocked `assessor2` from continue their review, and allo Reviewer2 to start one
- Log in again as `testAssigner1` and reassign back section 2 from `Reviewer2` to `assessor2`
  - Now if you log in as `assessor2` you can see the previous review is back and can be continued from where stopped

### Improvement
- Also added a better handling to Review subheader, so when is in progress, submitted or locked it show different messages. The discontinued in this case is also treated as locked:
<img width="732" alt="Screen Shot 2021-09-21 at 9 34 46 PM" src="https://user-images.githubusercontent.com/16461988/134148386-3a4f9684-e1e6-46bc-b559-4818f23491de.png">
<img width="733" alt="Screen Shot 2021-09-21 at 9 32 13 PM" src="https://user-images.githubusercontent.com/16461988/134148398-9e0b0861-03a3-4596-bd83-5696191442c6.png">
<img width="741" alt="Screen Shot 2021-09-21 at 9 34 38 PM" src="https://user-images.githubusercontent.com/16461988/134148403-e5ed1ec6-c51a-4bbb-b62b-5eaad488f25f.png">
